test('sympy/integrals/tests/test_integrals.py') fails on master ?

test('sympy/integrals/tests/test_integrals.py') seems to fail on master.
I'm running Python3.5, sympy-dev, MacOS 10.15.3

> there were a lot of places where Mul.is_integer wouldn't evaluate to True, even in a lot of obvious cases. hence the upgrade

It is expected that `is_integer` will not always resolve.

I tried the following diff from master:
```diff
diff --git a/sympy/core/mul.py b/sympy/core/mul.py
index 0bbd2fb67a..31fd0890d6 100644
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1256,9 +1256,7 @@ def _eval_is_integer(self):
 
         n, d = fraction(self)
         if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
+            pass
         if is_rational:
             if d is S.One:
                 return True
```
Running all of the core tests the only failure I got was
```
    def test_Mul_is_irrational():
        expr = Mul(1, 2, 3, evaluate=False)
        assert expr.is_irrational is False
        expr = Mul(1, I, I, evaluate=False)
>       assert expr.is_rational is True # I * I = -1
E       assert None is True
E        +  where None = I*I.is_rational
```
That particular test was changed in the commit e092acd leading to this issue.

Does anything else in the code depend on this?

--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -7,7 +7,7 @@ from .basic import Basic
 from .singleton import S
 from .operations import AssocOp
 from .cache import cacheit
-from .logic import fuzzy_not, _fuzzy_group
+from .logic import fuzzy_not, _fuzzy_group, fuzzy_and
 from .compatibility import reduce
 from .expr import Expr
 from .parameters import global_parameters
@@ -1249,28 +1249,25 @@ class Mul(Expr, AssocOp):
 
     def _eval_is_integer(self):
         from sympy import fraction
+        from sympy.core.numbers import Float
 
         is_rational = self._eval_is_rational()
         if is_rational is False:
             return False
 
-        n, d = fraction(self)
+        # use exact=True to avoid recomputing num or den
+        n, d = fraction(self, exact=True)
         if is_rational:
             if d is S.One:
                 return True
-            elif d == S(2):
+        if d.is_even:
+            if d.is_prime:  # literal or symbolic 2
                 return n.is_even
-        # if d is even -- 0 or not -- the
-        # result is not an integer
-        if n.is_odd and d.is_even:
-            return False
-        if not is_rational:
-            if n.has(d):
-                _self = Mul(n, Pow(d, -1))
-                if _self != self:
-                    return _self.is_integer
-            elif n.is_rational is True and d == S.One:
-                return True
+            if n.is_odd:
+                return False  # true even if d = 0
+        if n == d:
+            return fuzzy_and([not bool(self.atoms(Float)),
+            fuzzy_not(d.is_zero)])
 
     def _eval_is_polar(self):
         has_polar = any(arg.is_polar for arg in self.args)

--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1255,10 +1255,6 @@ class Mul(Expr, AssocOp):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
             if d is S.One:
                 return True
@@ -1268,6 +1264,13 @@ class Mul(Expr, AssocOp):
         # result is not an integer
         if n.is_odd and d.is_even:
             return False
+        if not is_rational:
+            if n.has(d):
+                _self = Mul(n, Pow(d, -1))
+                if _self != self:
+                    return _self.is_integer
+            elif n.is_rational is True and d == S.One:
+                return True
 
     def _eval_is_polar(self):
         has_polar = any(arg.is_polar for arg in self.args)

--- a/sympy/core/tests/test_arit.py
+++ b/sympy/core/tests/test_arit.py
@@ -357,6 +357,33 @@ def test_Mul_doesnt_expand_exp():
     assert sqrt(2)*2**Rational(1, 4)*5**Rational(3, 4) == 10**Rational(3, 4)
     assert (x**(-log(5)/log(3))*x)/(x*x**( - log(5)/log(3))) == sympify(1)
 
+def test_Mul_is_integer():
+
+    k = Symbol('k', integer=True)
+    n = Symbol('n', integer=True)
+    nr = Symbol('nr', rational=False)
+    nz = Symbol('nz', integer=True, zero=False)
+    nze = Symbol('nze', even=True, zero=False)
+    e = Symbol('e', even=True)
+    o = Symbol('o', odd=True)
+    i2 = Symbol('2', prime=True, even=True)
+
+    assert (k/3).is_integer is None
+    assert (nz/3).is_integer is None
+    assert (nr/3).is_integer is False
+    assert (x*k*n).is_integer is None
+    assert (e/o).is_integer is None
+    assert (o/e).is_integer is False
+    assert (o/i2).is_integer is False
+    assert Mul(o, 1/o, evaluate=False).is_integer is True
+    assert Mul(k, 1/k, evaluate=False).is_integer is None
+    assert Mul(nze, 1/nze, evaluate=False).is_integer is True
+    assert Mul(2., S.Half, evaluate=False).is_integer is False
+
+    s = 2**2**2**Pow(2, 1000, evaluate=False)
+    m = Mul(s, s, evaluate=False)
+    assert m.is_integer
+
 
 def test_Add_Mul_is_integer():
     x = Symbol('x')
@@ -366,19 +393,11 @@ def test_Add_Mul_is_integer():
     nk = Symbol('nk', integer=False)
     nr = Symbol('nr', rational=False)
     nz = Symbol('nz', integer=True, zero=False)
-    e = Symbol('e', even=True)
-    o = Symbol('e', odd=True)
 
     assert (-nk).is_integer is None
     assert (-nr).is_integer is False
     assert (2*k).is_integer is True
     assert (-k).is_integer is True
-    assert (k/3).is_integer is None
-    assert (nz/3).is_integer is None
-    assert (nr/3).is_integer is False
-    assert (x*k*n).is_integer is None
-    assert (e/o).is_integer is None
-    assert (o/e).is_integer is False
 
     assert (k + nk).is_integer is False
     assert (k + n).is_integer is True
@@ -1501,9 +1520,12 @@ def test_Mul_is_irrational():
     expr = Mul(1, 2, 3, evaluate=False)
     assert expr.is_irrational is False
     expr = Mul(1, I, I, evaluate=False)
-    assert expr.is_rational is True # I * I = -1
+    assert expr.is_rational is None # I * I = -1 but *no evaluation allowed*
+    # sqrt(2) * I * I = -sqrt(2) is irrational but
+    # this can't be determined without evaluating the
+    # expression and the eval_is routines shouldn't do that
     expr = Mul(sqrt(2), I, I, evaluate=False)
-    assert expr.is_irrational is not True
+    assert expr.is_irrational is None
 
 
 def test_issue_3531():

